dftimewolf/lib/collectors/timesketch.py

```diff
@@ -0,0 +1,259 @@
+      name: str | None = None,
+      critical: bool = False
+  ) -> None:
+    """"""
+    super(TimesketchSearchEventCollector, self).__init__(
+        state, name=name, critical=critical)
+    self.query_string: str = ''
+    self.start_datetime: datetime.datetime | None = None
+    self.end_datetime: datetime.datetime | None = None
+    self.indices: List[int] = []
+    self.labels: List[str] = []
+    self.output_format: str = ''
+    self.return_fields: str = ''
+    self.search_name: str = ''
+    self.search_description: str = ''
+    self.include_internal_columns: bool = False
+    self.sketch_id: int = 0
+    self.sketch: sketch.Sketch | None = None
+
+  # pylint: disable=arguments-differ,too-many-arguments
+  def SetUp(
+      self,
+      sketch_id: str,
+      query_string: str = '*',
+      start_datetime: datetime.datetime | None = None,
+      end_datetime: datetime.datetime | None = None,
+      indices: str | None = None,
+      labels: str | None = None,
+      output_format: str = 'pandas',
+      return_fields: str = '*',
+      search_name: str | None = None,
+      search_description: str | None = None,
+      include_internal_columns: bool = False,
+      token_password: str = '',
+      endpoint: str | None = None,
+      username: str | None = None,
+      password: str | None = None
+  ) -> None:
+    """Sets up the TimesketchSearchEventCollector.
+
+    Args:
+      sketch_id: the Timesketch sketch ID.
+      query_string: the query string.  Defaults to '*' (all events).
+      start_datetime: the start datetime.
+      end_datetime: the end datetime.
+      indices: the comma-separated Timesketch indices.
+      labels: the comma-separated Timesketch event labels.
+      output_format: the output format.  Defaults to 'pandas'.
+      return_fields: the return fields.  Defaults to '*'.
+      search_name: an optional name for the search.
+      search_description: an optional description for the search.
+      include_internal_columns: include Timesketch internal columns.  Defaults
+          to False.
+      token_password: optional password used to decrypt the
+          Timesketch credential storage. Defaults to an empty string since
+          the upstream library expects a string value. An empty string means
+          a password will be generated by the upstream library.
+      endpoint: Timesketch server URL (e.g. http://localhost:5000/).
+          Optional when token_password is provided.
+      username: Timesketch username. Optional when token_password is provided.
+      password: Timesketch password. Optional when token_password is provided.
+    """
+    if not sketch_id:
+      self.ModuleError('Sketch ID is not set.', critical=True)
+
+    if not start_datetime or not end_datetime:
+      self.ModuleError(
+          'Both the start and end datetime must be set.', critical=True)
+
+    if output_format not in _VALID_OUTPUT_FORMATS:
+      self.ModuleError('Output formats not valid.', critical=True)
```

Comment:
Can we specify which output formats are valid in the error message?

Reply:
Done